scripts/parse_performance_metrics.py

```
@@ -0,0 +1,200 @@
+        parsed_line = _try_decode_json(raw_line)
+        if not parsed_line:
+            continue
+        message = parsed_line.get("message")
+        if isinstance(message, str):
+            metrics = _extract_metrics_from_message(message)
+            if metrics is not None:
+                return metrics
+    raise RuntimeError(
+        "No performance metrics were found. Ensure the integration tests "
+        "printed a PERFORMANCE_METRICS line and the log was captured correctly.",
+    )
```

Ruff (0.14.0)

Avoid specifying long messages outside the exception class

(TRY003)